### PR TITLE
Introduce implementation for hook X509/Sni in icingadb

### PR DESCRIPTION
**Pull request by raviks789**
The hook X509/Sni is implemented for icingadb in this commit.

### CODE REVIEW

File: library/Icingadb/ProvidedHook/X509/Sni.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            'host_name'      => 'name',
+            'host_address'   => 'address',
+            'host_address6'  => 'address6'
+        ];
+
+        $queryHost = $queryHost->columns($hostStatusCols);
+
+        $this->applyRestrictions($queryHost);
+
+        $filter = Filter::where('name', 'dummy-1');
```
Comment (nilmerg) [inline library/Icingadb/ProvidedHook/X509/Sni.php:37]: You forgot to remove your test :wink:

File: library/Icingadb/ProvidedHook/X509/Sni.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        if ($filter !== null) {
+            $queryString = $filter->toQueryString();
+            $filterCondition = QueryString::parse($queryString);
+            FilterProcessor::apply($filterCondition, $queryHost);
```
Comment (nilmerg) [inline library/Icingadb/ProvidedHook/X509/Sni.php:41]: The `FilterProcessor` class is now not necessary anymore. Use this instead:

```suggestion
            $queryHost->filter($filterCondition);
```

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: library/Icingadb/ProvidedHook/X509/Sni.php (hunk 1 above)
Comment (raviks789) [reply library/Icingadb/ProvidedHook/X509/Sni.php:37]: Thank you, I will remove this test case.

File: library/Icingadb/ProvidedHook/X509/Sni.php (hunk 2 above)
Comment (raviks789) [reply library/Icingadb/ProvidedHook/X509/Sni.php:41]: It makes sense. I will make the changes.

**Review by nilmerg — APPROVED** [review]

**Pull request merged**